### PR TITLE
Don't show widevine install prompt when user doesn't want

### PR DESCRIPTION
**Pull request by simonhong**
User can turn it off from settings page or prompt dialog.

Fix https://github.com/brave/brave-browser/issues/5341

Screen Shot 2019-07-24 at 12 28 32 AM

Screen Shot 2019-07-25 at 9 04 52 AM

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
`yarn test --filter=WidevinePermissionRequestBrowserTest.*`

1. Start browser with clean profile
2. Load https://bitmovin.com/demos/drm
3. Check widevine install prompt is visible
4. Check `Don't ask again` checkbox in the prompt and click Block button
5. Reload and check prompt is not visible
6. Open newtab and goto settings
7. Check `Don't ask Widevine install` option is turned on
8. Turn it off
9. Reload https://bitmovin.com/demos/drm
10. Check widevine install prompt is visible

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/widevine/widevine_permission_request_browsertest.cc [hunk 1]
```diff
@@ -100,6 +100,24 @@ IN_PROC_BROWSER_TEST_F(WidevinePermissionRequestBrowserTest, VisibilityTest) {
...
+  // Check permission bubble is not visible when user turns it off.
+  observer.bubble_added_ = false;
+  DontAskWidevineInstall(GetActiveWebContents(), true);
+  EXPECT_TRUE(content::NavigateToURL(GetActiveWebContents(),
+                                     GURL("chrome://newtab/")));
+  drm_tab_helper->OnWidevineKeySystemAccessRequest();
+  content::RunAllTasksUntilIdle();
+  EXPECT_FALSE(observer.bubble_added_);
+
+    // Check permission bubble is visible when user turns it on.
```
Comment (iefremov) [inline browser/widevine/widevine_permission_request_browsertest.cc]: whitespace

File: browser/widevine/widevine_utils.cc [hunk 2]
```diff
@@ -120,3 +121,13 @@ void InstallBundleOrRestartBrowser() {
   }
 }
 #endif
+
+void DontAskWidevineInstall(content::WebContents* web_contents, bool dont_ask) {
+  Profile* profile = static_cast<Profile*>(web_contents->GetBrowserContext());
+  profile->GetPrefs()->SetBoolean(kDontAskWidevineInstall, dont_ask);
+}
+
+bool ShouldAskWidevineInstall(content::WebContents* web_contents) {
```
Comment (iefremov) [inline browser/widevine/widevine_utils.cc]: I'd prefer not to add these methods... This one is used only once, and it is even shorter to directly query prefs in that place

File: browser/widevine/widevine_permission_request_browsertest.cc (hunk 1 above)
Comment (simonhong) [reply browser/widevine/widevine_permission_request_browsertest.cc]: Done.

File: browser/widevine/widevine_utils.cc (hunk 2 above)
Comment (simonhong) [reply browser/widevine/widevine_utils.cc]: Yup, deleted.

File: app/brave_generated_resources.grd [hunk 3]
```diff
@@ -166,6 +166,12 @@
       <message name="IDS_WALLET_PERMISSION_REQUEST_TEXT_FRAGMENT" desc="Text fragment for wallet installation permission request.">
           Install Crypto Wallets
       </message>
+      <message name="IDS_WIDEVINE_DONT_ASK_AGAIN_CHECKBOX" desc="Checkbox for prevent showing widevine install prompt">
+        Don't ask again
+      </message>
+      <message name="IDS_SETTINGS_ASK_WIDEVINE_INSTALL_DESC" desc="Text fragment for asking widevine install or not">
+        Ask when a site wants to install Widevine on your computer
```
Comment (rebron) [inline app/brave_generated_resources.grd]: Minor nit to add a period "." at the end.  "Ask when a site wants to install Widevine on your computer."

Comment (simonhong) [reply app/brave_generated_resources.grd]: Done.

**Comment by simonhong** [comment]
Ping to code owner @bridiver

**Comment by simonhong** [comment]
kindly ping @bridiver

**Pull request merged**